bson/marshal.go

```diff
@@ -133,7 +134,7 @@ func MarshalValue(val interface{}) (Type, []byte, error) {
 		return 0, nil, err
 	}
 	typ := sw.Next(2)
-	return Type(typ[0]), sw.Bytes(), nil
+	return Type(typ[0]), slices.Clone(sw.Bytes()), nil
```

Comment:
The Go Driver currently supports a min Go version of 1.18, which does not have a `"slices"` package (it was added in Go 1.21). Instead, use the same clone logic from [Marshal](https://github.com/mongodb/mongo-go-driver/blob/a9eddec0e7cf4cf37b504d220a33226a3c28bf72/bson/marshal.go#L86).

```suggestion
	buf := append([]byte(nil), sw.Bytes()...)
	return Type(typ[0]), buf, nil
```

Reply:
Ah yes, already using Go 1.23. I updated the PR to copy in a pre Go 1.21 way.

Reply:
Thanks! The updated code in `MarshalValue` looks great!